src/pubsub/index.js

```diff
@@ -538,11 +559,15 @@ class PubsubBaseProtocol extends EventEmitter {
    * @returns {Promise<Message>}
    */
   _buildMessage (message) {
-    const msg = utils.normalizeOutRpcMessage(message)
-    if (this.signMessages) {
-      return signMessage(this.peerId, msg)
-    } else {
-      return message
+    const signaturePolicy = this.globalSignaturePolicy
+    switch (signaturePolicy) {
+      case SignaturePolicy.StrictSign:
+        message.from = this.peerId.toB58String()
+        message.seqno = utils.randomSeqno()
+        return signMessage(this.peerId, utils.normalizeOutRpcMessage(message))
+      case SignaturePolicy.StrictNoSign:
+        return utils.normalizeOutRpcMessage(message)
```

Comment:
Is this correct? Previously, we did not normalize the message if no sign.
Moreover, [normalizeOutRpcMessage](https://github.com/libp2p/js-libp2p-interfaces/blob/v0.6.0/src/pubsub/utils.js#L97) will create the from property from the previous from property (might throw?)